services/mailservers/tcp_ping.go

```diff
@@ -0,0 +1,41 @@
+package mailservers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/status-im/status-go/tcpping"
+)
+
+type PingQuery struct {
+	Addresses []string `json:"addresses"`
+	TimeoutMs int      `json:"timeoutMs"`
+}
+
+type PingResult struct {
+	Address string  `json:"address"`
+	Latency *int    `json:"latency"`
+	PingErr *string `json:"error"`
+}
+
+func fromPingResultToJSON(r tcpping.Result) PingResult {
```

Comment:
```suggestion
func toPingResult(r tcpping.Result) PingResult {
```
maybe? `fromPingResult` is odd when `PingResult` is returned :D

Reply:
k